backends-velox/src/main/scala/io/glutenproject/backendsapi/velox/VeloxTransformerApi.scala

```diff
@@ -35,6 +33,35 @@ import org.apache.spark.sql.execution.datasources.velox.DwrfFileFormat
 
 class VeloxTransformerApi extends ITransformerApi with Logging {
 
+  /**
+   * Add the validation for Velox unsupported or mismatched expressions with specific input type,
+   * such as Cast(ArrayType).
+   */
+  def doValidate(blacklist: Map[String, String], expr: Expression): Boolean = {
+    val value = blacklist.get(expr.prettyName.toLowerCase())
+    if (value.isEmpty) {
+      return true
+    }
+    val inputTypeName = value.get
+    if (inputTypeName.equals(VeloxExpressionUtil.EMPTY_TYPE)) {
+      return false
+    } else {
+      for (input <- expr.children) {
+        if (inputTypeName.equals(input.dataType.typeName)) {
+          return false
+        }
+      }
+    }
+    true
+  }
+
+  /**
+   * Do validate the expressions based on the specific backend blacklist,
+   * the existed expression will fall back to Vanilla Spark.
+   */
+  override def doValidate(expr: Expression): Boolean =
+    GlutenConfig.getSessionConf.isVeloxBackend && doValidate(VELOX_EXPR_BLACKLIST, expr)
```

Comment:
maybe it doesn't need to check whether it's velox backend？this api is only for velox .